packages/hooks/src/createAddHook.ts

```diff
@@ -1,29 +1,41 @@
 /**
  * Internal dependencies
  */
-import validateNamespace from './validateNamespace.js';
-import validateHookName from './validateHookName.js';
+import validateNamespace from './validateNamespace';
+import validateHookName from './validateHookName';
+import type { Callback, Hooks, StoreKey } from '.';
 
 /**
- * @callback AddHook
  *
  * Adds the hook to the appropriate hooks container.
- *
- * @param {string}               hookName      Name of hook to add
- * @param {string}               namespace     The unique namespace identifying the callback in the form `vendor/plugin/function`.
- * @param {import('.').Callback} callback      Function to call when the hook is run
- * @param {number}               [priority=10] Priority of this hook
  */
+export type AddHook = (
+	hookName: string,
+	namespace: string,
+	callback: Callback,
+	priority?: number
+) => void;
```

Comment:
Can we please retain the arguments' docs?

```suggestion
export type AddHook = (
	/**
	 * Name of the hook to add.
	 */
	hookName: string,
	/**
	 * The unique namespace identifying the callback in the form `vendor/plugin/function`.
	 */
	namespace: string,
	/**
	 * Function to call when the hook is run
	 */
	callback: Callback,
	/**
	 * Priority of this hook
	 */
	priority?: number
) => void;
```